.github/workflows/license_compliance.yml

```diff
@@ -0,0 +1,299 @@
+name: License Compliance
+
+on:
+  pull_request:
+    paths:
+      - "**/pyproject.toml"
+  # Since we test PRs, there is no need to run the workflow at each
+  # merge on `main`. Let's use a cron job instead.
+  schedule:
+    - cron: "0 0 * * *" # every day at midnight
+  # FIXME: remove the following
```

Comment:
Ran successfully here https://github.com/deepset-ai/haystack/actions/runs/4924813403 so we can remove this part before merging with main.